src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp

```diff
@@ -209,6 +227,58 @@ class InitMarkReduceOpPath : public pass::MatcherPass {
     }
 };
 
+/* MarkFloatingPointRange marks paths that involve Range operations with floating point output data types,
+ * as well as their users allowed for propagation. This pass is needed to prevent accuracy data loss
+ * in cases of high range generation, which could suffer due to lowered precision.
+ */
+class MarkFloatingPointRange : public pass::MatcherPass {
+public:
+    OPENVINO_RTTI("MarkFloatingPointRange", "0");
+    MarkFloatingPointRange() {
+        MATCHER_SCOPE(MarkFloatingPointRange);
+        auto exp_pattern = pattern::wrap_type<ov::op::v4::Range>();
+
+        // through this nodes
+        const std::shared_ptr<Node> range_propagating_nodes = pattern::wrap_type<ov::op::v0::Convert,
+                                                                                 ov::op::v1::Greater,
+                                                                                 ov::op::v1::GreaterEqual,
+                                                                                 ov::op::v1::Less,
+                                                                                 ov::op::v1::LessEqual,
+                                                                                 ov::op::v1::Reshape,
+                                                                                 ov::op::v4::Range,
+                                                                                 ov::op::v0::Squeeze,
+                                                                                 ov::op::v0::Unsqueeze>();
+
+        matcher_pass_callback callback = [=](pattern::Matcher& m) {
+            const auto& node = m.get_match_root();
+            if (!node)
+                return false;
+
+            auto range = as_type_ptr<ov::op::v4::Range>(node);
```

Comment:
do we really need to cast it to "Range"?

Reply:
Yes, as only the Range operation should start this sequential path marking